### ISSUE TITLE
Recipe #231: Transcripts, Captions, and Subtitles - General Considerations

### ISSUE THREAD

**Issue by glenrobson** [IIIF/trc#73]
## Links

 * Original Issue: https://github.com/IIIF/cookbook-recipes/issues/247
 * Pull Request: https://github.com/IIIF/cookbook-recipes/pull/248
 * Preview: https://preview.iiif.io/cookbook/transcript-meta-recipe/recipe/0231-transcript-meta-recipe/

## Background and Summary

In the [May 2021](https://github.com/IIIF/trc/milestone/22?closed=1) TRC we discussed the two recipes to do with transcriptions and captions with AV material. There was some confusion on what the differences were with each option and it was suggested that we create a meta recipe to help guide readers choose when they should use `rendering`, supplementing annotation of a file or full IIIF annotations. This Meta recipe is intended to be that guide to help readers choose which option they want by focusing on what they want the viewer to do with the information.  

The discussions in the May TRC have provoked a number of changes to other recipes, some which are in this TRC call but some others that will need to be submitted to later calls. With this in mind many of the links in this meta recipe to the different options don't currently match the description but this will be worked on. The status of the links are:

 * Newspaper links - this currently uses `seeAlso` but will be updated to use `rendering`. This recipe may also be split and will come to a future TRC for review.
 * [Providing Access to Transcripts with A/V Content][0017] - this will be reviewed in this TRC
 * Using Captions and Subtitles with Video Content - this will be reviewed in this TRC

Recipes that need to be written:
 * [A Side-by-side Transcript of a Video Recording][0253] - this recipe is to be written but will be the only example for option 2
 * [Transcription of image-based content][016]
 * [Using Annotations for Transcripts][0079].

## Voting and changes

We welcome comments on the recipe and as well as voting +1, confused face or -1 feel free to add comments to this issue. If this issue is approved then the author will take account of the comments before we merge the branch in to the master cookbook branch.

If the recipe is rejected by the TRC then we will make the changes requested and resubmit it to a future TRC meeting. If you feel that your comments are substantial enough that the recipe should be looked at again by the TRC after the changes have been made please vote -1 (thumbs down).

Changes to the recipe will only be made after the TRC voting process has concluded.

**Comment by aisaac** [comment]
For these who would try to access the recipe 0219 from the preview here (recipe 231), be careful, the link is wrong. In this recipe's preview the link is
https://preview.iiif.io/cookbook/transcript-meta-recipe/recipe/0219-using-caption-file/
while the version of the recipe submitted to TRC (as per https://github.com/IIIF/trc/issues/75) is 
https://preview.iiif.io/cookbook/0219-using-caption-file/recipe/0219-using-caption-file/

**Comment by aisaac** [comment]
Overall I like the direction it sets for the other recipes!

I must say that while reading this I had again ontological questions (and I'm afraid this goes back to the discussions we had in Slack ;-))
Considering that the manifest for "Captions and Subtitles" is identical to that of transcript case 2, the reader may wonder whether captions and subtitles are a kind of transcript. Especially since WebVTT files that are addressed in "captions and subtitles" can also be represented (indirectly) via annotations (this will be in recipe 079).
Should the recipe clarify this?
Maybe a cheap editorial trick to avoid this issue would be to rename the "Captions and Subtitles" into "Captions and Subtitles files for displaying on top of the video", since the main point of this section is to handle "original" WebVTT/TTML files following the usual convention for these files.

Finally this recipe mentions other recipes including some to be written in the future. But shouldn't 074 (Video with captions in multiple languages) mentioned too?

**Comment by tomcrane** [comment]
Approve generally but a little more (in line with Antoine)

I'd like to avoid anyone concluding that if they want to produce a user experience where the text is sitting on top of the video, like captions, they have to use a single file WebVTT and can't use W3C Web Annotations.

(Ideally they'd have the text content in *both* formats, for maximum accessibility AND maximum re-use in the IIIF world).

**Comment by tomcrane** [comment]
One other tweak -

> The motivation of the corresponding Annotation Page is supplementing. 

The annos have the motivation rather than the page.

**Comment by nfreire** [comment]
Aproved generally, but a possible way to make it clear that subtitles/captions can be provided as  W3C Web Annotations, could be to mention two implementations options under the section 'Captions and Subtitles': the current one about WebVTT files; and the option of using  Web Annotations with temporal fragments.

**Comment by zimeon** [comment]
I agree with @tomcrane and @nfreire that there should be mention of web annotations for subtitles/captions. I think there should also be a specific recipe on using web annotations to point to. We might note the caveat that one might want to link to WebVTT in parallel because of lack of client support for web annotation captions at present.

**Comment by mcwhitaker** [comment]
@zimeon Is it not the [Using Annotations for Transcripts][0079] recipe what you are asking for? It is under option #3 for transcripts.

**Comment by mcwhitaker** [comment]
@zimeon and should the reference be in the meta recipe or in recipe [Using Captions and Subtitles](https://preview.iiif.io/cookbook/transcript-meta-recipe/recipe/0219-using-caption-file/)?

**Comment by tpendragon** [comment]
I'm :+1:, but there's another option for subtitles in the case of streaming media (HLS, for instance, which I assume most folks will use as shipping a 30-minute file otherwise is an awful experience) you'd just ship the HLS manifest with the link to the WebVTT and follow that standard. Or are you saying we should still annotate the subtitles as given and somehow the viewer should know not to annotate them (I'm against this, if that's not clear)?

**Comment by nfreire** [comment]
@mcwhitaker 
In my opinion, since in the [Using Annotations for Transcripts][0079] recipe, the annotations have temporal fragments, it is aplicable to both Transcripts and Captions/Subtitles.
Maybe the title of recipe 79 can be changed to accommodate captions/subtitles. Then it could be a reference for both option #3 under transcripts, and for the section 'Captions and Subtitles' here in the meta-recipe.

**Comment by tomcrane** [comment]
> Ideally they'd have the text content in both formats, for maximum accessibility AND maximum re-use in the IIIF world

Thinking about this a bit more, and also from some chat recently in the IIIF accessibility slack channel - I think it should be a recommendation, a recipe, that if you have time-synchronised text for AV, then you should do all you can to expose **both** WebVTT and W3C Web Annotations, not one or the other.

 - It's easy to transform from one to the other, in both directions. There could even be a simple hosted transforming service so no excuses.
 - W3C Web Annos give the maximum _computational affordance_ to your text, and fit in with existing image-based annos. As well as encouraging adoption generally.
 - ...but, assistive technologies today don't know what web annos are. The job of a viewer is to expose the content to the browser    for maximum human affordance, which means giving your AV content links to WebVTT resources, so that the viewer, or whatever is rendering the web page, can make nice HTML5 accessible AV content.

**Comment by mcwhitaker** [comment]
@nfreire  I like the suggestion of making recipe [Using Annotations for Transcripts][0079]  for Time-based Text, which includes both transcripts and captions/subtitles, under option #3 (converting the text to multiple W2C Web Annotations), and mentioning that recipe both under Transcripts and under Captions and Subtitles. 

@tpendragon  I am not sure I understand what you are saying. But what is being suggested is to allow the webVTT file to be used as is and also to recommend exposure of the contents of that file transformed into W3C Web annotations.

**Comment by tpendragon** [comment]
> @tpendragon I am not sure I understand what you are saying. But what is being suggested is to allow the webVTT file to be used as is and also to recommend exposure of the contents of that file transformed into W3C Web annotations.

Sorry for not being clear. We may want to add another use case for streaming media here, eventually. We serve HLS files for instance (so users don't have to download the entire file), which requires Segmented WebVTT embedded in the HLS manifest. In that case I don't think I'd want to annotate the WebVTT file because it doesn't make sense in the context of the stream, and if a viewer tried to render those it'd be a bad experience. Have I missed a chat about this scenario?

**Comment by mcwhitaker** [comment]
@tpendragon  Got it. No, we have not discussed this scenario yet.

**Comment by glenrobson** [comment]
### Issue 73 (Recipe #231: Transcripts, Captions, and Subtitles - General Considerations)
  +1: 23 [awead cubap dlpierce emulatingkat glenrobson hadro jonhartzler jtweed julsraemy kirschbombe ksclarke markpatton mbennett-uoe mcwhitaker mixterj mposton-folger nfreire rentonsa shuddles thehabes tpendragon triplingual zimeon]
   0: 0 []
  -1: 0 []
  Not TRC: 1 [mapninja]
  Ineligible: 0 []

### Result: 23 / 23 = 1.00
Super majority is in favor, issue is approved

**Comment by mcwhitaker** [comment]
> Thinking about this a bit more, and also from some chat recently in the IIIF accessibility slack channel - I think it should be a recommendation, a recipe, that if you have time-synchronised text for AV, then you should do all you can to expose **both** WebVTT and W3C Web Annotations, not one or the other.
> 

@tomcrane  - at a recent cookbook authors meeting @robcast asked a good question: when we offer both options, how do we tell the client that the options are one and the same? In other words, how do we tell the client that only one of the options should be rendered? 

I think we are back to the conversation about motivations. @aisaac has repeatedly expressed interest in the `transcribing` motivation. Before we can work on the recipe for using annotations for timed text (https://github.com/IIIF/cookbook-recipes/issues/79), we need to actually resolve this question.
